roles/nfs_file_share/tasks/nfs_file_share.yml

```diff
@@ -20,13 +20,20 @@
 
 - name: Create NFS export
   shell: |
-    /usr/lpp/mmfs/bin/mmnfs export add {{ scale_protocols.mountpoint }}/{{ item.item }} --client "{{ compute_subnet_cidr }}(Access_Type=RW,SQUASH=no_root_squash)"
+    /usr/lpp/mmfs/bin/mmnfs export add {{ scale_protocols.mountpoint }}/{{ item.item }} --client "{{ compute_subnet_cidr }}(Access_Type=RW,SQUASH=root_squash)"
     sleep 10              #This is temporary fix for Dbus error.
   with_items: "{{ export_list_exists.results }}"
   register: created_nfs_export
   when: is_admin_node | default(false) == true and item.rc != 0
   run_once: true
 
+- name: Update permissions for directories in export list
+  shell: |
+    find {{ scale_protocols.mountpoint }}/{{ item.item }} -type d ! -path "{{ scale_protocols.mountpoint }}/{{ item.item }}/.snapshots" -exec chmod 777 {} +
+  with_items: "{{ export_list_exists.results }}"
```

Comment:
@Anand-Reddy7 do we have any common validation where we are checking` scale_protocols.mountpoint`  is defined or not.

Reply:
Yes @rajan-mis we have validation in the main.tf file. Please take a look here : https://github.com/Anand-Reddy7/ibm-spectrum-scale-install-infra/blob/anand_mount_permission/roles/nfs_file_share/tasks/main.yml#L5

FYI @digvijay-ukirde

Reply:
@rajan-mis defined "scale_protocols.mountpoint is defined" on when condition.